File: lib-content-scrapers/src/main/java/com/ustadmobile/lib/contentscrapers/khanacademy/TopicListResponse.java
package com.ustadmobile.lib.contentscrapers.khanacademy;

import java.util.List;

public class TopicListResponse {

    public ComponentData componentProps;

    public class ComponentData {

        public List<Modules> modules;

        public class Modules {

            public List<Domains> domains;

            public class Domains {

                public String href;

                public String icon;

                public String translatedTitle;

                public String identifier;

                public List<Domains> children;

            }
        }
    }
}

File: lib-content-scrapers/src/jvmMain/kotlin/com/ustadmobile/lib/contentscrapers/khanacademy/TopicListResponse.kt
package com.ustadmobile.lib.contentscrapers.khanacademy

class TopicListResponse {

    var componentProps: ComponentData? = null

    inner class ComponentData {

        var modules: List<Modules>? = null

        inner class Modules {

            var domains: List<Domains>? = null

            inner class Domains {

                var href: String? = null

                var icon: String? = null

                var translatedTitle: String? = null

                var identifier: String? = null

                var children: List<Domains>? = null

            }
        }
    }
}
